platform/api/aws/api.go

```diff
@@ -44,9 +46,15 @@ type Options struct {
 	// SecretKey is the optional secret key to use. It will override all other sources
 	SecretKey string
 
+	// AMI is the AWS AMI to launch EC2 instances with.
+	// If it is one of the special strings alpha|beta|stable, it will be resolved to an actual ID.
 	AMI           string
 	InstanceType  string
 	SecurityGroup string
+
+	resolvedAMILock sync.Mutex
```

Comment:
`Options` is for things that are passed in, and this doesn't really fit.  The mutex should go in `API`, and indeed so should `ResolvedAMI`, since it can only be safely accessed via a non-exported mutex.